Presentation.Web/Tests/it-system/tabs/LocalAdmin.Reference.System.e2e.spec.ts

```diff
@@ -1,86 +1,116 @@
+                    .then(() => expect(getEditButtonFromReference(referenceName).isPresent()).toBe(true))
+                    .then(() => getEditButtonFromReference(referenceName).click())
+                    .then(() => inputFields.referenceDocUrl.clear())
+                    .then(() => inputFields.referenceDocUrl.sendKeys(invalidUrl))
+                    .then(() => headerButtons.editSaveReference.click())
+                    .then(() => expect(getUrlFromReference(referenceName).isPresent()).toBeFalsy());
+            });
+
+        it("Can delete a reference in a IT-system",
+            () => {
+                var referenceName = createReferenceName();
+                var referenceId = createReferenceId();
+                var validUrl = generateValidUrl();
+
+                loginHelper.loginAsLocalAdmin()
+                    .then(() => refHelper.goToSpecificItSystemReferences(itSystemName))
+                    .then(() => refHelper.createReference(referenceName, validUrl, referenceId))
+                    .then(() => getDeleteButtonFromReference(referenceName).click())
+                    .then(() => browser.switchTo().alert().accept())
+                    .then(() => expect(getReferenceId(referenceId).isPresent()).toBeFalse());
+            });
     });
 
-describe("Can edit reference URL",
-    () => {
+function createItSystemName() {
+    return `ItSystemRefTest${new Date().getTime()}`;
+}
 
-        beforeAll(() => {
-            loginHelper.loginAsLocalAdmin();
-            refHelper.createReference(consts.refTitle, consts.validUrl, consts.refId);
-        });
+function createReferenceName() {
+    return `Reference${new Date().getTime()}`;
+}
 
-        beforeEach(() => {
-            homePage.getPage();
-            browser.waitForAngular();
-        });
-
-        afterAll(() => {
-            refHelper.deleteReference(consts.refId);
-            testFixture.cleanupState();
-        });
+function createReferenceId() {
+    return `ID${new Date().getTime()}`;
+}
 
+function generateValidUrl() {
+    return `http://www.${new Date().getTime()}.dk/`;
+}
 
-        it("Able to edit references", () => {
-            browser.wait(homePage.isEditReferenceLoaded(), waitUpTo.twentySeconds);
-            expect(headerButtons.editReference.isEnabled()).toBe(true);
-        });
+function generateInvalidUrl()
+{
+    return `${new Date().getTime()}.dk/`;
+}
 
-        it("Able to insert invalid url", () => {
-            browser.wait(homePage.isEditReferenceLoaded(), waitUpTo.twentySeconds);
-            headerButtons.editReference.click();
-            inputFields.referenceDocUrl.clear();
-            inputFields.referenceDocUrl.sendKeys(consts.invalidUrl);
-            headerButtons.editSaveReference.click();
+function getReferenceId(refName : string) {
+    return element(by.xpath('//*/tbody/*/td/a[text()="' + refName +'"]/parent::*/parent::*//td[@data-element-type="referenceIdObject"]'));
```

Comment:
Er vi sikre på at "referenceIdObject" kommer med ud i HTML'en på data-element-type - det gælder for editReferences, deleteReferences og referenceObject .. synes bare ikke jeg kan finde hvor for referenceIdObject?

Reply:
Disse bliver lavet når kendo grid bliver udfyldt.